actions/ulog_ingestion/src/ulog_ingestion/utils.py

```diff
@@ -0,0 +1,287 @@
+
+    Args:
+    - values: A list of tuples, each containing a field name, field type, and field value.
+
+    Returns:
+    - A dictionary representing the JSON object constructed from the input values.
+    """
+    json_instance = {}
+    array_values = {}
+
+    for field_name, field_type, field_value in values:
+        if "[" in field_name:
+            array_name = field_name.split("[")[0]
+            array_values.setdefault(array_name, []).append(
+                convert_value(field_type, field_value)
+            )
+        else:
+            json_instance[field_name] = convert_value(field_type, field_value)
+
+    json_instance.update(array_values)
+    return json_instance
+
+
+def convert_value(field_type: str, value: Any) -> Any:
+    """
+    Converts a field value to its corresponding JSON type.
+
+    Args:
+    - field_type: The type of the field as a string.
+    - value: The value to be converted.
+
+    Returns:
+    - The converted value in its appropriate JSON type.
+    """
+    json_type = TYPE_MAPPING.get(field_type, "string")
+    if json_type == "integer":
+        return int(value)
+    elif json_type == "number":
+        return float(value)
+    elif json_type == "boolean":
+        return bool(int(value))
+    elif json_type == "string":
+        return str(value)
+    else:
+        return value
+
+
+def create_message_path_records(topic: Any, field_data: Any) -> None:
+    """
+    Creates message path records for a given topic.
+
+    Args:
+    - topic: The topic object to which the message paths will be added.
+    - field_data: A list of field data objects containing the message definition.
+    """
+    array_list = list()
+    for field in field_data:
+        # For now only allow these types. TODO: Add support for nested types
+        if field.type_str in TYPE_MAPPING_CANONICAL.keys():
+            if "[" in field.field_name:
+                array_name = field.field_name.split("[")[0]
+                array_field_type = f"{field.type_str}[]"
+
+                if array_name not in array_list:
+                    print(
+                        f"Adding array: {array_name}, type: {array_field_type}, canonical: {topics.CanonicalDataType.Array}"
+                    )
+                    topic.add_message_path(
+                        request=topics.AddMessagePathRequest(
+                            message_path=array_name,
+                            data_type=array_field_type,  # TBD
+                            canonical_data_type=topics.CanonicalDataType.Array,
+                        )
+                    )
+
+                    print(
+                        f"Adding sub-field for array: {sub_array_name}, type: {field.type_str}, canonical: {TYPE_MAPPING_CANONICAL[field.type_str]}"
+                    )
+
+                    # Add another message path for the array elements
+                    sub_array_name = f"{array_name}.[*]"
```

Comment:
Do we also handle things like `sensor_msgs/Channels[]`, where the sub message paths should be the attributes of `sensor_msg/Channel` objects?